base/loading.jl

```diff
@@ -142,9 +453,10 @@ function _require_from_serialized(mod::Symbol, path_to_try::String, depmodnames:
                 depmods[i] = M
             end
         else
-            modpath = find_package(string(modname))
-            modpath === nothing && return ErrorException("Required dependency $modname not found in current path.")
-            mod = _require_search_from_serialized(modname, String(modpath))
+            info = find_package(into, string(modname))
```

Comment:
this (this thing that is being found) isn't being loaded into `into`, it's being loaded into `mod`

Reply:
but also `mod` and `modname` aren't Module names, they're UUIDs

Reply:
Except they're not actual UUIDs, they're the 64-bit integers you're calling "UUIDs". Also, why is `modname` called `modname` if it's not a module name? Can you throw me a bone here and tell me how to get the module associated with `mod` instead of just telling me what's wrong with the code?

Reply:
The `into` argument to `_require` is what's passed in here and as far as I can tell, it results in the correct module being passed in here. I think you may not be understanding what I'm trying to accomplish here. I've tried to explain and we've not gotten anywhere; this all works correctly in the absence of module precompilation, so if you could please make a RP on top of this one that fixes all the precompilation stuff, that would be great.

Reply:
As far as I can tell, `mod` is the name of the module that needs to be created. `into` is the module into which the binding for `mod` is to be created when the process is done. So it seems to me that `mod` is **not** the module that should be passed as the `into` argument here. Why do you think it is?